src/main/java/org/fenixedu/academic/domain/person/personIdentifier/PersonIdentifier.java

```diff
@@ -35,9 +36,11 @@ public void setType(PersonIdentifierType type) {
     @Override
     public void setIdentifier(String identifier) {
         String regexExpression = getType().getExpression();
+        LocalizedString expressionHelpMessage = getType().getExpressionHelpMessage();
 
         if (StringUtils.isNotBlank(regexExpression) && !identifier.matches(regexExpression)) {
-            throw new DomainException("error.person.personIdentifier.identifier.invalidFormat", getType().getName().getContent());
+            throw new DomainException("error.person.personIdentifier.identifier.invalidFormatWithHelpMessage",
+                    getType().getName().getContent(), expressionHelpMessage.getContent());
```

Comment:
This can provoke a NPE, for the existing data where `regexExpression` is filled but the `expressionHelpMesssage` isn't.